protocol/x/clob/keeper/deleveraging.go

```diff
@@ -137,13 +137,16 @@ func (k Keeper) GetInsuranceFundBalance(
 }
 
 // CanDeleverageSubaccount returns true if a subaccount can be deleveraged.
-// Specifically, this function returns true if both of the following are true:
-// - The subaccount's total net collateral is negative.
-// This function returns an error if `GetNetCollateralAndMarginRequirements` returns an error.
+// This function returns two booleans, canDeleverageSubaccount and shouldFinalSettlePosition.
+// - canDeleverageSubaccount is true if the subaccount can be deleveraged.
+// - shouldFinalSettlePosition is true if the position should be final settled (this occurs at oracle price)
+// This function returns an error if `GetNetCollateralAndMarginRequirements` returns an error or if there is
+// an error when fetching the clob pair for the provided perpetual.
 func (k Keeper) CanDeleverageSubaccount(
```

Comment:
Major change here is that this function now returns a `shouldFinalSettlePosition` boolean to tell the protocol whether or not the subaccount should be final settle deleveraged or regularly deleveraged